test/cases/testdata/withkeyword/test-withkeyword-1054.yaml

```diff
@@ -0,0 +1,21 @@
+cases:
+- data:
+  modules:
+    - |
+      package generated
+      
+      allow {
+        input.x > 0
+      }
+      
+      test_allow {
+        base_test_case := {"x": 1}
+      
+        allow with input as object.union(base_test_case, {"x": 2})
+        allow with input as object.union(base_test_case, object.union(base_test_case, {"x": 2}))
+        not allow with input as object.union(base_test_case, object.union(base_test_case, {"x": -1}))
+      }
+  note: withkeyword/rewrite declared variables nested in function call
```

Comment:
[nit] This captures the problem that's fixed here, but it would perhaps be cool to also assert that the right thing happens -- i.e. have `base_test_case := {"x": 0}` in line 12. Also, we could split these in 2-3 test cases instead.

Reply:
thanks for review srenatus, I pushed up a new commit with the changes